packages/react/polymorphic/src/forwardRefWithAs.ts

```diff
@@ -35,7 +53,7 @@ interface ForwardRefExoticComponentWithAs<
    * events are typed for consumers.
    */
   <As extends keyof JSX.IntrinsicElements>(
-    props: MergeWithDOMProps<As, OwnProps & { as: As }>
+    props: MergeWithDOMProps<As, ExtendedProps<E, P> & { as: As }>
```

Comment:
Why did we move to single letter generics? I quite liked how they were more descriptive before.

Reply:
Ah this is a habit of mine because I find that single letter communicates more clearly that it is using a generic and not some predefined type somewhere. I always get more confused when generics are named strangely. Happy to change it back though if it's easier for you. Will do that 🙂

Reply:
I can see arguments for both, I think in some of these it was quite useful though yeah.